virtual_ecosystem/models/soil/soil_model.py

```diff
@@ -208,15 +220,19 @@ def integrate(self) -> dict[str, DataArray]:
             [
                 self.data[name].to_numpy()
                 for name in map(str, self.data.data.keys())
-                if name.startswith("soil_c_pool_") or name.startswith("soil_enzyme_")
+                if name.startswith("soil_c_pool_")
+                or name.startswith("soil_enzyme_")
+                or str(name).startswith("soil_n_pool_")
             ]
         )
 
         # Find and store order of pools
         delta_pools_ordered = {
             name: np.array([])
             for name in map(str, self.data.data.keys())
-            if name.startswith("soil_c_pool_") or name.startswith("soil_enzyme_")
+            if name.startswith("soil_c_pool_")
+            or name.startswith("soil_enzyme_")
+            or str(name).startswith("soil_n_pool_")
```

Comment:
This should not be needed, if it is not needed for the others. 
```suggestion
            if name.startswith("soil_c_pool_")
            or name.startswith("soil_enzyme_")
            or name.startswith("soil_n_pool_")
```